mobile/src/main/java/org/openhab/habdroid/background/BackgroundUtils.java

```diff
@@ -0,0 +1,169 @@
+                    new NotificationChannel(CHANNEL_ID_BACKGROUND, name, importance);
+            channel.setDescription(description);
+            notificationManager.createNotificationChannel(channel);
+
+            name = context.getString(R.string.notification_channel_background_error);
+            description =
+                    context.getString(R.string.notification_channel_background_error_description);
+            importance = NotificationManager.IMPORTANCE_DEFAULT;
+            channel = new NotificationChannel(CHANNEL_ID_BACKGROUND_ERROR, name, importance);
+            channel.setDescription(description);
+            channel.setLightColor(ContextCompat.getColor(context, R.color.openhab_orange));
+            channel.enableLights(true);
+            notificationManager.createNotificationChannel(channel);
+        }
+    }
+
+    public static Notification makeBackgroundNotification(Context context,
+            @StringRes int msgId, @DrawableRes int iconId, boolean isError,
+            NotificationCompat.Action action) {
+        return makeBackgroundNotification(context, context.getString(msgId), iconId, isError,
+                action);
+    }
+
+    /**
+     * Makes notification for background tasks. Sets notification channel, importance and more
+     * depending on the parameters.
+     *
+     * @param context
+     * @param msg Message to show.
+     * @param iconId Icon to show.
+     * @param isError true if it's an error.
+     * @param action Action to show as a button, e.g. "Retry"
+     * @return
+     */
+    public static Notification makeBackgroundNotification(Context context, String msg,
+            @DrawableRes int iconId, boolean isError, NotificationCompat.Action action) {
+        Intent notificationIntent = new Intent(context, MainActivity.class);
+        notificationIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP
+                | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        PendingIntent intent = PendingIntent.getActivity(context, 0,
+                notificationIntent, 0);
+
+        NotificationCompat.Builder nb = new NotificationCompat.Builder(context,
+                isError ? CHANNEL_ID_BACKGROUND_ERROR : CHANNEL_ID_BACKGROUND)
+                .setLargeIcon(drawableToBitmap(context.getResources().getDrawable(iconId)))
+                .setSmallIcon(R.drawable.ic_openhab_appicon_white_24dp)
+                .setContentTitle(context.getString(R.string.app_name))
+                .setWhen(System.currentTimeMillis())
+                .setStyle(new NotificationCompat.BigTextStyle().bigText(msg))
+                .setContentText(msg)
+                .setAutoCancel(true)
+                .setContentIntent(intent)
+                .setColor(ContextCompat.getColor(context, R.color.openhab_orange))
+                .setCategory(isError ? NotificationCompat.CATEGORY_ERROR
+                        : NotificationCompat.CATEGORY_PROGRESS)
+                .setOngoing(!isError)
```

Comment:
IMHO error notifications aren't ongoing either, as
- they don't represent an ongoing action
- there's no reason to prevent them from being swiped away

Reply:
Errors aren't set to ongoing here. It's the "waiting for network" notification.

Reply:
Oops, right. The connection to the notification type is a bit on-obvious here though...maybe make this explicit through a parameter to the function?

Reply:
I would like keep the function signature as small as possible. If I add everything where `isError` is used to the signature, it'll add a lot of duplicate code to the function's call.

Reply:
Do you want to create a generic framework or not? ;-) If it's supposed to be generic it must not cater to specific behavior IMHO.
At the least the Javadoc needs to point out that isError == false means the notification reason is expected to be an ongoing process.